R/antimalarial_resistance.R

```diff
@@ -0,0 +1,149 @@
+  dt_slow_parasite_clearance <- rep(parameters$dt, length = length(drugs))
+  
+  for(i in seq_along(parameters$antimalarial_resistance_drug)) {
+    drug <- parameters$antimalarial_resistance_drug[[i]]
+    treated_with_drug <- which(drugs == drug)
+    resistance_timestep <- match_timestep(ts = parameters$antimalarial_resistance_timesteps[[i]], t = timestep)
+    artemisinin_resistance_proportion[treated_with_drug] <- parameters$prop_artemisinin_resistant[[i]][resistance_timestep]
+    partner_drug_resistance_proportion[treated_with_drug] <- parameters$prop_partner_drug_resistant[[i]][resistance_timestep]
+    slow_parasite_clearance_probability[treated_with_drug] <- parameters$slow_parasite_clearance_prob[[i]][resistance_timestep]
+    early_treatment_failure_probability[treated_with_drug] <- parameters$early_treatment_failure_prob[[i]][resistance_timestep]
+    late_clinical_failure_probability[treated_with_drug] <- parameters$late_clinical_failure_prob[[i]][resistance_timestep]
+    late_parasitological_failure_probability[treated_with_drug] <- parameters$late_parasitological_failure_prob[[i]][resistance_timestep]
+    reinfection_during_prophylaxis_probability[treated_with_drug] <- parameters$reinfection_during_prophylaxis[[i]][resistance_timestep]
+    dt_slow_parasite_clearance[treated_with_drug] <- parameters$dt_slow_parasite_clearance[[i]]
+  }
+  
+  resistance_parameters <- list()
+  resistance_parameters$artemisinin_resistance_proportion <- artemisinin_resistance_proportion
+  resistance_parameters$partner_drug_resistance_proportion <- partner_drug_resistance_proportion
+  resistance_parameters$slow_parasite_clearance_probability <- slow_parasite_clearance_probability
+  resistance_parameters$early_treatment_failure_probability <- early_treatment_failure_probability
+  resistance_parameters$late_clinical_failure_probability <- late_clinical_failure_probability
+  resistance_parameters$late_parasitological_failure_probability <- late_parasitological_failure_probability
+  resistance_parameters$reinfection_during_prophylaxis_probability <- reinfection_during_prophylaxis_probability
+  resistance_parameters$dt_slow_parasite_clearance <- dt_slow_parasite_clearance
+  
+  if(any(c(length(resistance_parameters$artemisinin_resistance_proportion),
+           length(resistance_parameters$partner_drug_resistance_proportion), 
+           length(resistance_parameters$slow_parasite_clearance_probability), 
+           length(resistance_parameters$early_treatment_failure_probability), 
+           length(resistance_parameters$late_clinical_failure_probability), 
+           length(resistance_parameters$late_parasitological_failure_probability), 
+           length(resistance_parameters$reinfection_during_prophylaxis_probability),
+           length(resistance_parameters$dt_slow_parasite_clearance)) != length(drugs))) {
+    stop("Length of one or more resistance parameter vectors does not match length of drugs vector")
+  }
```

Comment:
This check seems like overkill to me?
You specifiy that those vectors are `length(drugs)` by definition when you create them. Is there reason why they should/would change?

Reply:
I don't think they should/would, but I added the test as insurance. You're probably right though, so have removed this check from the function.